### PR TITLE
manifest reader unit tests

### PR DESCRIPTION
**Pull request by RomanLevkovych**
Implemented tests for `feature/machine-manifest-update` branch.

### CODE REVIEW

File: applications/ExecutionManager/include/manifest_reader.hpp [hunk 1]
```diff
@@ -54,7 +56,12 @@ class ManifestReader : public IManifestReader
   /**
    * @brief Hardcoded path to folder with adaptive applications.
    */
-  const static std::string corePath;
+  const std::string corePath =
+      #ifndef UNIT_TEST
```
Comment (adklimkiew) [inline applications/ExecutionManager/include/manifest_reader.hpp]: This is hack, not allowed

File: applications/ExecutionManager/include/manifest_reader.hpp [hunk 2]
```diff
@@ -42,7 +42,9 @@ class ManifestReader : public IManifestReader
 
   ~ManifestReader() override = default;
 
+#ifndef UNIT_TEST
```
Comment (adklimkiew) [inline applications/ExecutionManager/include/manifest_reader.hpp]: This sort of hacking i.e. polluting production code with solutions for easier testing, is not allowed. You should test your class only by public interface

File: applications/ExecutionManager/include/manifest_reader.hpp (hunk 1 above)
Comment (adklimkiew) [reply applications/ExecutionManager/include/manifest_reader.hpp]: Find better way to control and set path string

File: applications/ExecutionManager/include/manifests.hpp [hunk 3]
```diff
@@ -29,6 +29,13 @@ struct ProcessName
 {
   std::string applicationName;
   std::string processName;
+
+  bool operator==(const ProcessName& procName) const
```
Comment (adklimkiew) [inline applications/ExecutionManager/include/manifests.hpp]: in such cases rhs is used as a name and in this context it is clear

File: applications/ExecutionManager/src/manifest_reader.cpp [hunk 4]
```diff
@@ -62,7 +60,11 @@ ManifestReader::getStatesSupportedByApplication()
 std::vector<MachineState> ManifestReader::getMachineStates()
 {
   static const std::string manifestPath =
+    #ifdef UNIT_TEST
```
Comment (adklimkiew) [inline applications/ExecutionManager/src/manifest_reader.cpp]: Not allowed

File: applications/ExecutionManager/test/em_test.cpp [hunk 5]
```diff
@@ -1,76 +1,250 @@
...
+    output << applicationManifestForPath;
+  }
+}
+
+/// INIT ENVIRONMENT
+::testing::Environment* const manifestReaderEnvironment =
+    ::testing::AddGlobalTestEnvironment(new ManifestReaderEnvironment);
+/// -------------------
+
+TEST(ManifestReaderTest, getApplicationListFail)
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: Write testing class, all reusable code should go there, in particular environment should be its member, and use TEST_F for all your tests

File: applications/ExecutionManager/test/em_test.cpp [hunk 6]
```diff
@@ -1,76 +1,250 @@
...
+                       + "/manifest.json";
+    std::ofstream output{path};
+
+    json applicationManifestForPath = appManifest.second;
+    output << applicationManifestForPath;
+  }
+}
+
+/// INIT ENVIRONMENT
+::testing::Environment* const manifestReaderEnvironment =
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: Do you need this object to be on the heap?

File: applications/ExecutionManager/test/em_test.cpp [hunk 7]
```diff
@@ -1,76 +1,250 @@
...
+    ::testing::AddGlobalTestEnvironment(new ManifestReaderEnvironment);
+/// -------------------
+
+TEST(ManifestReaderTest, getApplicationListFail)
+{
+  // remove folder with data
+  manifestReaderEnvironment->TearDown();
+
+  ManifestReader reader;
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: Should be a member of testing class

File: applications/ExecutionManager/test/em_test.cpp [hunk 8]
```diff
@@ -1,76 +1,250 @@
...
+
+/// INIT ENVIRONMENT
+::testing::Environment* const manifestReaderEnvironment =
+    ::testing::AddGlobalTestEnvironment(new ManifestReaderEnvironment);
+/// -------------------
+
+TEST(ManifestReaderTest, getApplicationListFail)
+{
+  // remove folder with data
+  manifestReaderEnvironment->TearDown();
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: Should be hidden

File: applications/ExecutionManager/test/em_test.cpp [hunk 9]
```diff
@@ -1,76 +1,250 @@
...
+
+  ManifestReader reader;
+  ASSERT_THROW(reader.getListOfApplications(),
+               std::runtime_error);
+
+  // create folder with data
+  manifestReaderEnvironment->SetUp();
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: Should be hidden

File: applications/ExecutionManager/test/em_test.cpp [hunk 10]
```diff
@@ -1,76 +1,250 @@
...
+  ASSERT_EQ(manifSerialized, returnObj);
+}
+
+TEST(ManifestReaderTest, getMachineStates)
+{
+  ManifestReader reader;
+
+  const std::vector<MachineState> expectedOutput
+      = {"Startup", "Running", "Shutdown"};
+  const auto& result = reader.getMachineStates();
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: This local variable not needed

File: applications/ExecutionManager/test/em_test.cpp [hunk 11]
```diff
@@ -1,76 +1,250 @@
...
+
+  ManifestReader reader;
+  ASSERT_THROW(reader.getListOfApplications(),
+               std::runtime_error);
+
+  // create folder with data
+  manifestReaderEnvironment->SetUp();
+}
+
+TEST(ManifestReaderTest, getJsonDataTest)
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: getJsonData should be private. So this test will disappear

File: applications/ExecutionManager/test/em_test.cpp [hunk 12]
```diff
@@ -1,76 +1,250 @@
...
+  json manifSerialized = ManifestReaderTestData::machineManifest;
+
+    std::ofstream output;
+    output.open(ManifestReaderTestData::machineManifestPath);
+    output << manifSerialized;
+}
+
+void ManifestReaderEnvironment::createDirs()
+{
+    std::string createDir = "mkdir ";
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: const

File: applications/ExecutionManager/test/em_test.cpp (hunk 5 above)
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: Test name shoukld follow convention: Should_ExpectedBehavior_When_StateUnderTest, for example:
ShouldThrowExceptionWhenAgeLessThan18
ShouldFailToWithdrawMoneyForInvalidAccount

File: applications/ExecutionManager/test/em_test.cpp (hunk 6 above)
Comment (RomanLevkovych) [reply applications/ExecutionManager/test/em_test.cpp]: Yes, it is said in docs, that it should be allocated in heap

File: applications/ExecutionManager/test/em_test.cpp [hunk 13]
```diff
@@ -156,6 +156,70 @@ TEST_F(ManifestReaderTest, ShouldFaildWhenDirectoryNotExists)
   EXPECT_THROW(reader.getStatesSupportedByApplication(), std::runtime_error);
 }
 
+TEST_F(ManifestReaderTest, ShouldOtherKeysThanMachineStateWhenParse)
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: Sth is missing in test name

File: applications/ExecutionManager/test/em_test.cpp [hunk 14]
```diff
@@ -156,6 +156,70 @@ TEST_F(ManifestReaderTest, ShouldFaildWhenDirectoryNotExists)
...
+{
+  std::string redundantMachineManifest =
+      R"({"Machine_manifest":{"Machine_manifest_id":"Machine",)"
+      R"("Mode_declaration_group":[{"Function_group_name":"MachineState",)"
+      R"("Mode_declarations":[{"Mode":"Startup"},{"Mode":"Running"},)"
+      R"({"Mode":"Shutdown"}]},{"Function_group_name":"TestGroupName",)"
+      R"("Mode_declarations":[{"Mode":"tes1"},{"Mode":"tes2"}]}]}})";
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: You use two ways for creating manifest content.  But only one should be used consistently

File: applications/ExecutionManager/test/em_test.cpp [hunk 15]
```diff
@@ -156,6 +156,70 @@ TEST_F(ManifestReaderTest, ShouldFaildWhenDirectoryNotExists)
...
+
+  std::ofstream testOutput{testConf.machineManifestPath};
+  testOutput << redundantMachineManifest;
+  testOutput.close();
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: This code repeats in tests -> private function can be created

File: applications/ExecutionManager/test/em_test.cpp [hunk 16]
```diff
@@ -156,6 +156,70 @@ TEST_F(ManifestReaderTest, ShouldFaildWhenDirectoryNotExists)
   EXPECT_THROW(reader.getStatesSupportedByApplication(), std::runtime_error);
 }
 
+TEST_F(ManifestReaderTest, ShouldOtherKeysThanMachineStateWhenParse)
+{
+  std::string redundantMachineManifest =
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: This is not redundant manifest...

File: applications/ExecutionManager/test/em_test.cpp [hunk 17]
```diff
@@ -171,6 +235,54 @@ TEST_F(ManifestReaderTest, ShouldReturnMapOfApplicationsForStates)
...
+              expectedResult[appsForState.first].size());
+
+    const auto& expResultForState = expectedResult[appsForState.first];
+
+    for (size_t i = 0; i < expResultForState.size(); i++)
+    {
+      ASSERT_EQ(appsForState.second.at(i).processName,
+                expResultForState.at(i).processName);
+      ASSERT_EQ(appsForState.second.at(i).applicationName,
+                expResultForState.at(i).applicationName);
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: Is this not possible to compare directly two std maps?

File: applications/ExecutionManager/src/manifest_reader.cpp [hunk 18]
```diff
@@ -58,29 +57,30 @@ ManifestReader::getStatesSupportedByApplication()
...
   return res;
 }
 
 std::vector<MachineState> ManifestReader::getMachineStates()
 {
   json manifestData = getJsonData(conf.machineManifestPath);
-
   MachineManifest manifest = manifestData.get<MachineManifest>();
-
   std::vector<MachineState> res;
```
Comment (adklimkiew) [inline applications/ExecutionManager/src/manifest_reader.cpp]: Move this vector creation (potentially expensive) to the place of actual usage

File: applications/ExecutionManager/test/em_test.cpp [hunk 19]
```diff
@@ -156,6 +156,70 @@ TEST_F(ManifestReaderTest, ShouldFaildWhenDirectoryNotExists)
...
+
+  std::ofstream testOutput{conf.machineManifestPath};
+  testOutput << redundantMachineManifest;
+  testOutput.close();
+
+
+  ManifestReader reader{conf};
+  auto res = reader.getMachineStates();
+  ASSERT_EQ(res,
+            (std::vector<MachineState>{}));
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: You could create private const empty vector to be used as expected value in tests

File: applications/ExecutionManager/src/manifest_reader.cpp (hunk 18 above)
Comment (adklimkiew) [reply applications/ExecutionManager/src/manifest_reader.cpp]: and reserve

File: applications/ExecutionManager/test/em_test.cpp (hunk 17 above)
Comment (RomanLevkovych) [reply applications/ExecutionManager/test/em_test.cpp]: unfortunately, no

Comment (adklimkiew) [reply applications/ExecutionManager/test/em_test.cpp]: Why not? What are compile error messages? Maybe you miss operator== for ProcessName

File: applications/ExecutionManager/include/manifest_reader.hpp [hunk 20]
```diff
@@ -12,13 +12,25 @@ namespace ExecutionManager
...
 using MachineState = std::string;
+
+/**
+ * @brief Config of Manifest Reader: main consts that will be
+ *        injected externally.
+ */
+struct ManifestReaderConf
+{
+  const std::string corePath{"./bin/applications"};
+  const std::string machineManifestPath{ "../applications/ExecutionManager/machine_manifest.json"};
```
Comment (adklimkiew) [inline applications/ExecutionManager/include/manifest_reader.hpp]: This if file name

File: applications/ExecutionManager/include/manifest_reader.hpp [hunk 21]
```diff
@@ -12,13 +12,25 @@ namespace ExecutionManager
...
 
 using MachineState = std::string;
+
+/**
+ * @brief Config of Manifest Reader: main consts that will be
+ *        injected externally.
+ */
+struct ManifestReaderConf
+{
+  const std::string corePath{"./bin/applications"};
```
Comment (adklimkiew) [inline applications/ExecutionManager/include/manifest_reader.hpp]: Is this related to application manifests or sth else? Name should inform about it

File: applications/ExecutionManager/include/manifest_reader.hpp [hunk 22]
```diff
@@ -12,13 +12,25 @@ namespace ExecutionManager
 {
 
 using MachineState = std::string;
+
+/**
+ * @brief Config of Manifest Reader: main consts that will be
+ *        injected externally.
+ */
```
Comment (adklimkiew) [inline applications/ExecutionManager/include/manifest_reader.hpp]: Remove that comment

File: applications/ExecutionManager/include/manifest_reader.hpp (hunk 20 above)
Comment (RomanLevkovych) [reply applications/ExecutionManager/include/manifest_reader.hpp]: Yes, it is a path to file with machine manifest.

File: applications/ExecutionManager/test/em_test.cpp [hunk 23]
```diff
@@ -1,76 +1,290 @@
...
 
-/**
- * @brief Check Application Manifest serialization and deserialization.
- */
-TEST(SerializationTest, applicationManifestTest)
+
+class ManifestReaderTests : public ::testing::Test
+{
+protected:
+  void SetUp() override;
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: set up/tear down functions are called for each test, but only some of them use these default settings. Therefore these functions should be parametrized with ManifetRaderConf, AppManifests string, and MachineManifest string and some other data if required. And renamed to installManifests / removeManifests or similar

File: applications/ExecutionManager/test/em_test.cpp [hunk 24]
```diff
@@ -1,76 +1,290 @@
...
+  for (auto& appsForState: result)
+  {
+    ASSERT_EQ(appsForState.second.size(),
+              expectedResult[appsForState.first].size());
+
+    const auto& expResultForState = expectedResult[appsForState.first];
+
+    for (size_t i = 0; i < expResultForState.size(); i++)
+    {
+      ASSERT_EQ(appsForState.second.at(i).processName,
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: Compare two maps directly

File: applications/ExecutionManager/test/em_test.cpp [hunk 25]
```diff
@@ -1,76 +1,290 @@
...
+                expResultForState.at(i).applicationName);
+    }
+  }
+}
+
+TEST_F(ManifestReaderTests, ShouldEmptyMapWhenNoProvided)
+{
+  ManifestReaderConf conf{"./empty-manifest/", ""};
+
+  system(("mkdir -p " + conf.corePathToApplicationsFolder + "/app").c_str());
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: This low-level code should disappear

File: applications/ExecutionManager/test/em_test.cpp [hunk 26]
```diff
@@ -1,76 +1,290 @@
...
+  ManifestReader reader{conf};
+  auto result = reader.getStatesSupportedByApplication();
 
-  MachineManifest deserializedManifest = json::parse(testJsonResult);
+  ASSERT_EQ(result,
+            (std::map<MachineState, std::vector<ProcessName>>{}));
 
-  ASSERT_EQ(manifestSerialized.dump(), testJsonResult);
-  ASSERT_EQ(deserializedManifest, manifestSerialized);
+  system(("rm -rf " + conf.corePathToApplicationsFolder).c_str());
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: This low-level code should disappear

File: applications/ExecutionManager/test/em_test.cpp (hunk 24 above)
Comment (RomanLevkovych) [reply applications/ExecutionManager/test/em_test.cpp]: Direct comparison doesn't work for the maps.

Comment (RomanLevkovych) [reply applications/ExecutionManager/test/em_test.cpp]: Direct comparison doesn't work for maps.

Comment (adklimkiew) [reply applications/ExecutionManager/test/em_test.cpp]: Of course it works

Comment (RomanLevkovych) [reply applications/ExecutionManager/test/em_test.cpp]: Yes, it works, but when I use it to compare expected and actual result, I receive that objects in vectors are different by their memory location

Comment (RomanLevkovych) [reply applications/ExecutionManager/test/em_test.cpp]: Well, it works, just my inattentiveness :[

Comment (adklimkiew) [reply applications/ExecutionManager/test/em_test.cpp]: You need to understand why that happens. Maybe print both connections and compare them.. They should be equal and compared by values, not addresses.

Comment (adklimkiew) [reply applications/ExecutionManager/test/em_test.cpp]: Ok, great! I'm curious what was the reason for difference?

Comment (RomanLevkovych) [reply applications/ExecutionManager/test/em_test.cpp]: In `ProcessInfo::operator==` I used `rhs.processName` twice instead of `rhs.applicationName` and `rhs.processName` respectively

File: applications/ExecutionManager/include/manifest_reader.hpp [hunk 27]
```diff
@@ -13,13 +13,9 @@ namespace ExecutionManager
...
 using MachineState = std::string;
 
-/**
- * @brief Config of Manifest Reader: main consts that will be
- *        injected externally.
- */
 struct ManifestReaderConf
 {
-  const std::string corePath{"./bin/applications"};
+  const std::string corePathToApplicationsFolder{"./bin/applications"};
```
Comment (adklimkiew) [inline applications/ExecutionManager/include/manifest_reader.hpp]: appManifestPath

File: applications/ExecutionManager/include/manifest_reader.hpp [hunk 28]
```diff
@@ -13,13 +13,9 @@ namespace ExecutionManager
...
   const std::string machineManifestPath{ "../applications/ExecutionManager/machine_manifest.json"};
```
Comment (adklimkiew) [inline applications/ExecutionManager/include/manifest_reader.hpp]: machineManifestFile

File: applications/ExecutionManager/test/em_test.cpp [hunk 29]
```diff
@@ -49,222 +95,282 @@ void ManifestReaderTests::writeJson(const std::string &content, const std::strin
...
-  ManifestReader reader{testConf};
-
-  auto res = reader.getMachineStates();
-
-  ASSERT_EQ(res,
-            (std::vector<MachineState>{"Startup", "Running", "Shutdown"}));
-}
-
-TEST_F(ManifestReaderTests, ShouldFaildWhenDirectoryNotExists)
+const std::vector<ManifestReaderTestConf> machineManifestTestsConfigs =
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: It is hard to read that structure. Maybe split this vector into elements and name them properly

File: applications/ExecutionManager/test/em_test.cpp [hunk 30]
```diff
@@ -49,222 +95,282 @@ void ManifestReaderTests::writeJson(const std::string &content, const std::strin
...
 {
-  ManifestReader reader{ManifestReaderConf{"./not-exists", "manifest"}};
+  ManifestReaderTestConf{
+    {"msm"},
+    {},
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: What is the meaning of empty brackets {}?

File: applications/ExecutionManager/test/em_test.cpp [hunk 31]
```diff
@@ -49,222 +95,282 @@ void ManifestReaderTests::writeJson(const std::string &content, const std::strin
...
+    MachineManifest{
+      MachineManifestInternal{
+        "Machine",
+        {
+          ModeDeclarationGroup{
+            "MachineState",
+            {Mode{"Startup"}, Mode{"Running"}, Mode{"Shutdown"}}
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: You dont need to use Startup, Running & Shutdown modes. You can use whatever names come to your mind

File: applications/ExecutionManager/test/em_test.cpp [hunk 32]
```diff
@@ -49,222 +95,282 @@ void ManifestReaderTests::writeJson(const std::string &content, const std::strin
...
 
-  writeJson(machineManifestWithUnsupportedGroup, testConf.machineManifestPath);
-
-  ManifestReader reader{testConf};
   auto res = reader.getMachineStates();
 
-  ASSERT_EQ(res,
-            (std::vector<MachineState>{"Startup", "Running", "Shutdown"}));
-
+  if (GetParam().machineManifest.manifest.modeDeclarationGroups.size())
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: Testing code should have lowest cyclomatic complexity (as simple as possible) i.e. no branches, no if statements, no nested constructs etc

File: applications/ExecutionManager/test/em_test.cpp [hunk 33]
```diff
@@ -49,222 +95,282 @@ void ManifestReaderTests::writeJson(const std::string &content, const std::strin
...
-  auto res = reader.getMachineStates();
-  ASSERT_EQ(res, emptyMachineStates);
+  ManifestReader reader{ManifestReaderConf{"./not-exists", "manifest"}};
 
+  EXPECT_THROW(reader.getStatesSupportedByApplication(), std::runtime_error);
 }
 
-
-TEST_F(ManifestReaderTests, ShouldEmptyReturnWhenEmptyMachineStates)
+const std::vector<ManifestReaderTestConf> applicationManifestReadintTestConf =
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: Very long and unclear data structure. Split it into elements. Correct Readint

File: applications/ExecutionManager/test/em_test.cpp [hunk 34]
```diff
@@ -49,222 +95,282 @@ void ManifestReaderTests::writeJson(const std::string &content, const std::strin
...
-  ManifestReader reader{conf};
-  auto result = reader.getStatesSupportedByApplication();
-
-  ASSERT_EQ(result,
-            (std::map<MachineState, std::vector<ProcessInfo>>{}));
-
-  system(("rm -rf " + conf.corePathToApplicationsFolder).c_str());
+    EXPECT_EQ(result, availableAppsForStates);
+  }
+  else
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: No branches in testing code

File: applications/ExecutionManager/test/em_test.cpp [hunk 35]
```diff
@@ -9,38 +9,84 @@
 using namespace ExecutionManager;
 using nlohmann::json;
 
+struct ManifestReaderTestConf
```
Comment (adklimkiew) [inline applications/ExecutionManager/test/em_test.cpp]: use two sepearate test structst for machine manifest and  app manifests as it seems they are completely separated functionalities. Therefore I think about splitting manifest reader into two classes for simplicity...

File: applications/ExecutionManager/test/em_test.cpp (hunk 30 above)
Comment (RomanLevkovych) [reply applications/ExecutionManager/test/em_test.cpp]: It is application manifest struct

Comment (RomanLevkovych) [reply applications/ExecutionManager/test/em_test.cpp]: I omit those manifests that I don't use

File: applications/ExecutionManager/test/em_test.cpp (hunk 35 above)
Comment (RomanLevkovych) [reply applications/ExecutionManager/test/em_test.cpp]: In order to save current api. we may leave our current `ManifestReader` class as adapter for other parts, but internally it can encapsulate classes for Application and Manifest parsing and we can test them separately.

File: applications/ExecutionManager/test/em_test.cpp (hunk 31 above)
Comment (RomanLevkovych) [reply applications/ExecutionManager/test/em_test.cpp]: Yeah, since I used structs from my code names haven't changed.

File: applications/ExecutionManager/test/em_test.cpp (hunk 30 above)
Comment (adklimkiew) [reply applications/ExecutionManager/test/em_test.cpp]: It is not easy to deduce it

File: applications/ExecutionManager/test/em_test.cpp (hunk 35 above)
Comment (adklimkiew) [reply applications/ExecutionManager/test/em_test.cpp]: Anyway test conf struct must be divided into two

File: applications/ExecutionManager/include/execution_manager.hpp [hunk 36]
```diff
@@ -94,6 +98,7 @@ class ExecutionManager
                           const std::vector<ProcessInfo>&);
 private:
 
+private:
```
Comment (rBekesh) [inline applications/ExecutionManager/include/execution_manager.hpp]: not needed

File: applications/ExecutionManager/include/execution_manager.hpp [hunk 37]
```diff
@@ -50,6 +52,8 @@ class ExecutionManager
 
   bool setMachineState(pid_t processId, std::string state);
 private:
+  using ApplicationState = ::ApplicationStateManagement::ApplicationState;
```
Comment (rBekesh) [inline applications/ExecutionManager/include/execution_manager.hpp]: not needed

File: applications/ExecutionManager/include/execution_manager.hpp [hunk 38]
```diff
@@ -50,6 +52,8 @@ class ExecutionManager
...
+  using StateError = ::MachineStateManagement::StateError;
```
Comment (rBekesh) [inline applications/ExecutionManager/include/execution_manager.hpp]: not needed

File: applications/ExecutionManager/include/manifest_reader.hpp [hunk 39]
```diff
@@ -10,13 +10,21 @@ namespace ExecutionManager
 {
 
 using MachineState = std::string;
+
+struct ManifestReaderConf
+{
+  const std::string pathToApplicationsFolder{"./bin/applications"};
+  const std::string machineManifestFilePath{ "../applications/ExecutionManager/machine_manifest.json"};
```
Comment (rBekesh) [inline applications/ExecutionManager/include/manifest_reader.hpp]: remove extra space

File: applications/ExecutionManager/test/em_test.cpp [hunk 40]
```diff
@@ -1,76 +1,664 @@
...
+{
+  const auto& testParam = GetParam();
+  ManifestReader reader{testParam.conf};
+
+  auto result = reader.getStatesSupportedByApplication();
+
+    EXPECT_EQ(result, availableAppsForStates);
+}
+
+
```
Comment (rBekesh) [inline applications/ExecutionManager/test/em_test.cpp]: remove empty line

File: applications/ExecutionManager/test/em_test.cpp [hunk 41]
```diff
@@ -1,76 +1,664 @@
...
+       ShouldReturnAvailableMachineStatesWhenProvided)
+{
+  ManifestReader reader{GetParam().conf};
+
+  auto res = reader.getMachineStates();
+
+  ASSERT_EQ(res, availableMachineStates);
+}
+
+
```
Comment (rBekesh) [inline applications/ExecutionManager/test/em_test.cpp]: remove empty line

File: applications/ExecutionManager/test/em_test.cpp [hunk 42]
```diff
@@ -1,76 +1,664 @@
...
+   {"app"}
+};
+
+const std::vector<MachineManifestReaderTestConf> machineManifestTestsConfigs =
+{
+  regularMachineManifestTestConf,
+  machineManifestWithAdditionalFunctionGroup,
+};
+
+const std::vector<ApplicationManifestReaderTestConf> applicationManifestReadintTestConf =
```
Comment (rBekesh) [inline applications/ExecutionManager/test/em_test.cpp]: applicationManifestReadingTestConf

File: applications/ExecutionManager/test/em_test.cpp [hunk 43]
```diff
@@ -1,76 +1,664 @@
...
+  commandLineOptionShortForm,
+  commandLineOptionLongForm,
+  commandLineOptionSimpleForm,
+  commandLineOptionShortFormWithoutArg,
+  commandLineOptionLongFormWithoutArg
+};
+
+INSTANTIATE_TEST_SUITE_P(ApplicationManifestTests,
+                         ApplicationManifestReadingTests,
+                         ::testing::ValuesIn(applicationManifestReadintTestConf));
```
Comment (rBekesh) [inline applications/ExecutionManager/test/em_test.cpp]: applicationManifestReadingTestConf
